docs/runtime-contract.md

```diff
@@ -125,47 +112,36 @@ the OCI specification as long as:
 - Platforms MAY provide mechanisms for post-mortem viewing of filesystem
   contents from a particular execution. Because containers (particularly failing
   containers) can experience frequent starts, operators or platform providers
-  SHOULD limit the total space consumed by these failures.
-- A container should write its own termination message to `/dev/termination-log`
-  by default. If no message is written by the container, the last few lines of
-  log output should be reported as the execution error (i.e. by
-  [setting the `terminationMessagePolicy` to `FallbackToLogsOnError`](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message))
-  on Kubernetes.
-
-### Warnings
-
-As specified by OCI.
+  MAY limit the total space consumed by these failures.
 
 ### Operations
 
 [The OCI interface](https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc3/runtime.md#operations)
 SHOULD NOT be exposed within the container. The operator or platform provider
-MAY have the ability to directly interact with the OCI interface, but that is
-beyond the scope of this specification.
+MAY have the ability to directly interact with the OCI interface.
 
 An OPTIONAL method of invoking the `kill` operation MAY be exposed to developers
 to provide signalling to the container.
 
-### Hooks
-
-Operation hooks SHOULD NOT be configurable by the Knative developer. Operators
-or platform providers MAY use hooks to implement their own lifecycle controls.
-
 ### Linux Runtime
 
 #### File descriptors
 
-A read from the `stdin` file descriptor on the container SHOULD always result in
-`EOF`. The `stdout` and `stderr` file descriptors on the container SHOULD be
-collected and retained in a developer-accessible logging repository.
+A read from the `stdin` file descriptor on the container [SHOULD](https://github.com/knative/serving/blob/master/test/conformance/file_descriptor_test.go)
+always result in `EOF`. The `stdout` and `stderr` file descriptors on the container
+MUST be collected and retained in a developer-accessible logging repository.
```

Comment:
It would be nice to have a minimum logging requirement that is common across all Knative installs. Right now both /var/log and stdout are SHOULD which means that a developer need to go hunting through the docs or configuration for each Knative install to understand how logs are collected. My understanding is that right now in Knative Serving we collect stdout and stderr for all containers so we are already doing this.